messages/java/Makefile

```diff
@@ -2,4 +2,24 @@ include default.mk
 
 JSONSCHEMAS = $(shell find ../jsonschema -name "*.json")
 
-.built: $(JSONSCHEMAS)
+.codegen: src/main/java/io/cucumber/messages/Messages.java
+
+src/main/java/io/cucumber/messages/Messages.java: $(JSONSCHEMAS) ../jsonschema/scripts/codegen.rb ../jsonschema/scripts/templates/java.java.erb ../jsonschema/scripts/templates/java.enum.java.erb
+	echo "package io.cucumber.messages;" > $@
+	echo >> $@
+	echo "import java.util.ArrayList;" >> $@
+	echo "import java.util.Objects;" >> $@
+	echo "import java.util.Optional;" >> $@
+	echo >> $@
+	echo "import static java.util.Collections.unmodifiableList;" >> $@
+	echo "import static java.util.Objects.requireNonNull;" >> $@
+	echo >> $@
+	echo "// Generated code" >> $@
+	echo "@SuppressWarnings(\"unused\")" >> $@
+	echo "public final class Messages {" >> $@
```

Comment:
Is it a limitation of the code generator that all classes must be inside a single file? I think it would be better if they're all in the messages package.

Reply:
Yes, generating a single file is currently a limitation. What's the problem with a single file?

Reply:
A few  reasons:

0. No real reason to use an inner class that acts like a package.
1. Editors don't like big files. 
2. The `import io.cucumber.messages.Messages.Something` is stuttering.
3. Smaller change compared to the previous API. So fewer imports to fix while migrating.
4. We're currently flip flopping between using inner classes (protobuf, this PR) and upper level classes ( as generated by the maven plugin). It would be nice to be consistent.